sdk/analytics/src/main/java/com/klaviyo/analytics/networking/requests/ApiRequest.kt

```diff
@@ -0,0 +1,65 @@
+package com.klaviyo.analytics.networking.requests
+
+import java.net.URL
+
+/**
+ * Immutable representation of the data of a network request
+ */
+interface ApiRequest {
```

Comment:
I did something similar. I’m thinking it might be interesting though to more events like flushing the queue etc…though at the same time might be harder to find what you are looking for in the test app unless we add filter.

Reply:
Yeah I wouldn't be surprised if we have some edits to this once we start using it in the tester app.